src/commands/dev/gcs/watch.rs

```diff
@@ -23,7 +23,12 @@ pub fn watch_for_changes(
         let target = target.clone();
         commands::build(&target)?;
 
+        // acquire the lock
         let mut preview_id = preview_id.lock().unwrap();
+
+        // while holding the lock, assign a new preview id
+        // this allows the server to route subsequent requests
+        // to the proper script
```

Comment:
i think somewhere else in the code you explain what "the lock" is but it's probably worth explaining why there's a lock here, too; another instance of "why > what"